platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/GuestClaimsHandlerExt.java

```diff
@@ -13,129 +13,208 @@
+            List<String> immutableClaims, String profilesDirectory, String profilesBannerDirectory,
+            String availableClaimsFile) {
+
+        this.propertiesFileSupport = propertiesFileSupport;
+        this.immutableClaims = immutableClaims;
+        this.profilesDirectory = profilesDirectory;
+        this.profilesBannerDirectory = profilesBannerDirectory;
+        this.availableClaimsFile = availableClaimsFile;
 
-    private List<String> convertMapToList(Map<String, String> map) {
-        List<String> list = new ArrayList<>();
-        for (Map.Entry<String, String> entry : map.entrySet()) {
-            list.add(entry.getKey() + "=" + entry.getValue());
-        }
-        return list;
+        selectedClaimsProfileName = null;
     }
 
-    private List<String> loadAvailableClaims(String filePath) {
-        List<String> claimsList = new ArrayList<>();
-        if (!StringUtils.isEmpty(filePath)) {
-            File file = new File(filePath);
-            if (file.exists()) {
-                claimsList.addAll(loadPropertiesFile(file).keySet());
+    /**
+     * Called by the container to initialize the object.
+     */
+    public void init() {
+        Consumer<Map<String, String>> loggingConsumer = map -> {
+            if (!map.containsKey(PROFILE_NAME_KEY)) {
+                LOGGER.warn("Ignoring properties file without a profile name");
             }
-        }
-        return claimsList;
+        };
+
+        availableClaimsMap = propertiesFileSupport.loadSinglePropertiesFile(availableClaimsFile);
+
+        claimsProfileMaps = propertiesFileSupport.loadPropertiesFilesInDirectory(profilesDirectory)
+                .stream()
+                .peek(loggingConsumer)
+                .filter(map -> map.containsKey(PROFILE_NAME_KEY))
+                .collect(Collectors.toList());
+
+        claimsProfileBannerMaps = propertiesFileSupport.loadPropertiesFilesInDirectory(
+                profilesBannerDirectory)
+                .stream()
+                .peek(loggingConsumer)
+                .filter(map -> map.containsKey(PROFILE_NAME_KEY))
+                .collect(Collectors.toList());
     }
 
-    private Map<String, String> loadPropertiesFile(File propFile) {
-        Map<String, String> propertyMap = new HashMap<>();
-        if (propFile != null && propFile.exists()) {
-            Properties properties = PropertiesLoader.loadProperties(propFile.getAbsolutePath());
-            propertyMap = PropertiesLoader.toMap(properties);
+    /**
+     * Submit the name of the profile listed in each configuration to select that configuration
+     * for use. Applies retroactively for system-high and banner markings.
+     *
+     * @param selectedClaimsProfileName the name of the profile to be used.
+     */
+    public void setSelectedClaimsProfileName(String selectedClaimsProfileName) {
+        Map<String, Object> claimsProfiles = extractProfileNamePreserveResult(claimsProfileMaps);
```

Comment:
It won't be called often in the first place. All this is strictly for installation.